### ISSUE TITLE
[Bug] Can't call view function from aptos typescript sdk

### ISSUE THREAD

**Issue by 0xhsy** [aptos-labs/aptos-core#6418]
# 🐛 Bug

I compiled it using the #[view] annotation at the top of get_message using the code below. 
[https://github.com/aptos-labs/aptos-core/tree/main/aptos-move/move-examples/hello_blockchain](https://github.com/aptos-labs/aptos-core/tree/main/aptos-move/move-examples/hello_blockchain)
```
// hello_blockchain.move
#[view]
public fun get_message(addr: address): string::String acquires MessageHolder {
    assert!(exists<MessageHolder>(addr), error::not_found(ENO_MESSAGE));
    *&borrow_global<MessageHolder>(addr).message
}

// Move.toml
[package]
name = "Examples"
version = "0.0.0"

[addresses]
hello_blockchain = "0xef493b8f6a79ecb401a7bc2f8c5f37c0e69313f84045ada10bae643341dd8eff"

[dependencies]
AptosFramework = { git = "https://github.com/aptos-labs/aptos-core.git", subdir = "aptos-move/framework/aptos-framework/", rev = "aptos-node-v1.2.0" }


// aptos cli version 1.0.4
$ aptos move compile --bytecode-version 6
```
And I succeeded by deploying using aptos typescript sdk 1.6.0 and executing the set_message function.
[https://explorer.aptoslabs.com/txn/419039528](https://explorer.aptoslabs.com/txn/419039528)

**Finally, I call the get_message function as below, but I get an error. What could be the problem?**

```
export async function viewFunction(account: string, moduleName: string, functionName: string, chainId: string, typeArg: [], param: any) {
  const aptosClient = new AptosClient(aptosNodeUrl(chainId));

  const payload = {
    function: account + "::" + moduleName + "::" + functionName,
    type_arguments: typeArg,
    arguments: param,
  };

  log.debug(payload)
  // {
  //   arguments: ['0xef493b8f6a79ecb401a7bc2f8c5f37c0e69313f84045ada10bae643341dd8eff'],
  //   function: "0xef493b8f6a79ecb401a7bc2f8c5f37c0e69313f84045ada10bae643341dd8eff::message::get_message",
  //   type_arguments: []
  // }

  return await aptosClient.view(payload);
}
```

**ERROR message : Uncaught (in promise) Error: {"message":"PartialVMError with status INVALID_MAIN_FUNCTION_SIGNATURE and message function not marked as view function","error_code":"invalid_input","vm_error_code":null}**

**Comment by banool** [comment]
Taking a look.

**Comment by banool** [comment]
Okay I just repro'd this with the latest version of the CLI.

First, pull aptos-core and switch to this branch: `git push --set-upstream origin banool/testing_view_function`.

Then, make an account and publish the Move module:
```
~/a/core/target/debug/aptos init
~/a/core/target/debug/aptos move publish --named-addresses hello_blockchain=`yq .profiles.default.account < .aptos/config.yaml` --bytecode-version 6
```

Then update the addresses in "view function testing" in src/aptos_client.test.ts.

Then run this:
```
pnpm test -t "view function testing"
```

You'll get this error:
```
{"message":"PartialVMError with status INVALID_MAIN_FUNCTION_SIGNATURE and message function not marked as view function","error_code":"invalid_input","vm_error_code":null}
```

Trying to figure out why this isn't working now.

**Comment by movekevin** [comment]
I found the issue. If you always include --bytecode-version, the issue would not happen. This is only reproducible if:
1. You first call aptos move compile or aptos move publish with --bytecode-version 6 but this fails for some reason (e.g. compiler errors).
2. You then call aptos move publish WITHOUT --bytecode-version.
There's a bug in the CLI where an env variable is set in (1) to tell the compiler to generate bytecode v6 that's not cleared after (1) failed. (2) then use this but the package metadata (which indicates which functions in the code are view functions) defaults to v5 in (2) => The error above happens

**Comment by 0xhsy** [comment]
It was resolved by updating the aptos cli version to 1.0.5.
